src/Compilers/Core/MSBuildTask/ManagedCompiler.cs

```diff
@@ -312,6 +331,16 @@ protected override Encoding StandardOutputEncoding
 
         protected override int ExecuteTool(string pathToTool, string responseFileCommands, string commandLineCommands)
         {
+            if (ProvideCommandLineInvocation)
+            {
+                CommandLineInvocation = GenerateResponseFileContents();
```

Comment:
C++ claims to have recently done this (and I imagine they scale even larger than C#). But I don't know the details.
But yes, the large string risk is why I have it "off by default".
Alternatively, if we can produce a `string[]` of arguments I'm happy to consume that. I hope that is almost as easy as getting the full string.